### DISCUSSION TITLE
Add support for macOS native instead of Mac Catalyst

### DISCUSSION THREAD

**Discussion by tounaobun** [dotnet/maui#22765] [Ideas] [answered]
### Description

Mac Catalyst is great, but it has many restrictions using UIKit. Why not add a new branch for supporting native macOS.

### Public API Changes

..

### Intended Use-Case

..

**Comment by asi-evin** [comment]
I second this. Things like serial port are much simpler.

**Comment by ghost** [comment]
We've added this issue to our backlog, and we will work to address it as time and resources allow. If you have any additional information or questions about this issue, please leave a comment. For additional info about issue management, please read our [Triage Process](https://github.com/dotnet/maui/blob/main/docs/TriageProcess.md).

**Comment by jfversluis** [comment]
To be honest I don't think we'll be doing this anytime soon. I'll leave it open for now to see if it gains any traction. If not, we might close it after a while. Thanks for the suggestion!

**Comment by vnraju** [comment]
We have a .NET MAUI app which does encryption and decryption of files/folders using passwords and/or certificates.
Our app is not able to send mails or access the cert store.
I have tested with a normal C# cli app on MacOS, and it is able to access the cert store but not the .NET MAUI mac-catalyst app.

**Comment by shivamsaviant** [comment]
@jfversluis I have an application which support all the platforms (android, iOS, WinUI) including mac. It supports BLE device communication on all platforms and HID device communication on WinUI currently. We have started the process of adding HID device support for mac and now blocked as it is not being supported by macCatalyst. Any help would be appreciated. Thanks.

**Comment by maxibyte** [comment]
MacCatalyst makes a lot of things much more annoying than they need to be. Official support would, of course, be really great, and I think there’s a real need for it. Which is a real shame, because we were able to implement everything with MAUI for Windows, but the Mac version is currently just taking up too much time. On top of that, with Blazor Hybrid, we’re seeing noticeably poorer performance on MacCatalyst. Anyone who wants to use MAUI to develop a desktop app will choose a different framework once they realize that only MacCatalyst is supported. @vnraju @shivamsaviant Maybe the [mauiplatforms ](https://github.com/shinyorg/mauiplatforms ) project can help some people with their maccatalyst issues.

**Comment by mattleibow** [comment, marked as answer]
Wanted to let folks on this thread know there's now something concrete to try. Over in **[dotnet/maui-labs](https://github.com/dotnet/maui-labs)** we've been building experimental platform backends, including a **GTK4 backend for Linux** and a **native macOS AppKit backend** (macOS without Mac Catalyst). Both also ship a `BlazorWebView` host — via **WebKitGTK** on Linux and **WKWebView** on macOS — so the "run your .NET business logic natively + a Blazor UI in a native WebView" scenario works on each.

**Linux (GTK4)** — [README + setup](https://github.com/dotnet/maui-labs/blob/main/platforms/Linux.Gtk4/README.md):
- [`Microsoft.Maui.Platforms.Linux.Gtk4`](https://www.nuget.org/packages/Microsoft.Maui.Platforms.Linux.Gtk4) — the backend
- [`Microsoft.Maui.Platforms.Linux.Gtk4.BlazorWebView`](https://www.nuget.org/packages/Microsoft.Maui.Platforms.Linux.Gtk4.BlazorWebView) — BlazorWebView on WebKitGTK
- [`Microsoft.Maui.Platforms.Linux.Gtk4.Essentials`](https://www.nuget.org/packages/Microsoft.Maui.Platforms.Linux.Gtk4.Essentials) — Essentials APIs
- [`Microsoft.Maui.Platforms.Linux.Gtk4.Templates`](https://www.nuget.org/packages/Microsoft.Maui.Platforms.Linux.Gtk4.Templates) — `dotnet new` templates

**macOS (AppKit)** — [README + setup](https://github.com/dotnet/maui-labs/blob/main/platforms/MacOS/README.md):
- [`Microsoft.Maui.Platforms.MacOS`](https://www.nuget.org/packages/Microsoft.Maui.Platforms.MacOS) — the backend
- [`Microsoft.Maui.Platforms.MacOS.BlazorWebView`](https://www.nuget.org/packages/Microsoft.Maui.Platforms.MacOS.BlazorWebView) — BlazorWebView on WKWebView
- [`Microsoft.Maui.Platforms.MacOS.Essentials`](https://www.nuget.org/packages/Microsoft.Maui.Platforms.MacOS.Essentials) — Essentials APIs

All published as `0.1.0-preview` — prerelease, so add `--prerelease` when installing.

**Status — please read before building on it:**

- These backends are **experimental**.
- They are **not supported** — explicitly **not** covered by the [.NET MAUI Support Policy](https://dotnet.microsoft.com/platform/support/policy/maui), with **no support commitment** of any kind (no SLAs, no security-fix guarantees, no compatibility guarantees).
- There is **no roadmap to make them officially supported** platforms. Linux and macOS are not announced as supported .NET MAUI targets, and nothing here should be read as a plan or commitment to do so.
- APIs may change or break at any time, and a backend could be discontinued.

Building these has also been a good exercise of the extensibility work tracked in [#34099](https://github.com/dotnet/maui/issues/34099) — the places MAUI can stop actively preventing third-party platforms, without the team having to own or officially support them.

Give the experimental backends a try, file issues in [maui-labs](https://github.com/dotnet/maui-labs/issues), and let us know what's working and what's missing — with the understanding above about where things stand.
